docs: add page on log messages (#25930)
ethereum/go-ethereum: docs/_interface/logs.md

---
title: Geth logs
sort key: N
---

A Geth node continually reports messages to the console allowing users to monitor Geth's current 
status in real-time. The logs indicate when Geth is running normally and indicates when some 
attention is required. However, reading these logs can be difficult for new users. This page 
will help to interpret the log messages to better understand what Geth is doing.

Note that there are a large number of log messages covering a wide range of possible scenarios for 
a Geth node. This page will only address a subset of commonly seen messages. For more, see the 
[Geth Github](https://github.com/ethereum/go-ethereum), [Discord](https://discord.gg/WHNkYDsAKU) 
or search on [ethereum.stackexchange](https://ethereum.stackexchange.com/). Log messages are 

> usually sufficiently self-descrining that they do not require additional explanation.

Reviewer: ```suggestion
usually sufficiently self-describing that they do not require additional explanation.
```

starts downloading state data. This state data is stored in the form of a [Patricia Merkle Trie](https://ethereum.org/en/developers/docs/data-structures-and-encoding/patricia-merkle-trie/). Only the leaves of the trie are downloaded, 
the full trie structure is then locally regenerated from the leaves up. Meanwhile, the blockchain 
continues to progress and the target header is updated. This means some of the regenerated state 
data needs to be updated. This is known as *healing*.

Assuming Geth has a synced consensus client and some peers it will start importing headers, block 
bodies and receipts. The log messages for data downloading look as follows:

```
INFO [07-28|10:29:49.681] Block synchronisation started
INFO [07-28|10:29:50.427] Imported new block headers               count=1    elapsed=253.434ms number=12,914,945 hash=ee1a08..9ce38a
INFO [07-28|10:30:00.224] Imported new block receipts              count=64   elapsed=13.703s   number=12,914,881 hash=fef964..d789fc age=18m5s     size=7.69MiB
INFO [07-28|10:30:18.658] Imported new block headers               count=1    elapsed=46.715ms  number=12,914,946 hash=7b24c8..2d8006

> INFO [07-28|10:30:21.665] Imported new state entries

Reviewer 2: The "Imported new state entries" log is incomplete, missing the arguments.

> INFO [07-28|10:30:21.665] Imported new state entries

Reviewer 3: ```suggestion
```

```

> For state sync, Geth reports when the state heal is in progress. This can takea long time. 

Reviewer 2: ```suggestion
For state sync, Geth reports when the state heal is in progress. This can take a long time. 
```

that needs to be downloaded to heal the trie can increase as well as decrease over time. Ultimately, 
the state should heal faster than the blockchain progresses so the node can get in sync. When the state 
healing is finished there is a post-sync snapshot generation phase. The node is not in sync until the 
state healing phase is over. If the node is still regularly reporting `State heal in progress` it is not 
yet in sync - the state healing is still ongoing.

```
INFO [07-28|10:30:21.965] State heal in progress                   accounts=169,633@7.48MiB  slots=57314@4.17MiB    codes=4895@38.14MiB nodes=43,293,196@11.70GiB pending=112,626
INFO [09-06|01:31:59.885] Rebuilding state snapshot
INFO [09-06|01:31:59.910] Resuming state snapshot generation root=bc64d4..fc1edd accounts=0 slots=0 storage=0.00B dangling=0 elapsed=18.838ms
```

The sync can be confirmed using [`eth.syncing`](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_syncing)
- it will return `false` if the node is in sync. If `eth.syncing` returns anything other than `false` it has not 
finished syncing. Generally, if syncing is still ongoing, `eth.syncing` will return block info that looks as follows:

> ```json

Reviewer 2: ```suggestion
```console
```

With JSON type it shows red rectangles around all the words because it's invalid JSON

Author: ```suggestion
```js
```
for consistency with other snippets

asked about especially frequently on the [Geth Github](https://github.com/ethereum/go-ethereum) and 
[Discord](https://discord.gg/WHNkYDsAKU) channel.  

```sh
WARN [10-03|18:00:40.413] Unexpected trienode heal packet          peer=9f0e8fbf         reqid=6,915,308,639,612,522,441
```

The above is often seen and misinterpreted as a problem with snap sync. In reality, it indicates a request 
timeout that may be because I/O speed is low. It is usually not an issue, but if this message is seen very 
often over prolonged periods of time it might be rooted in a local connectivity or hardware issue.

```sh
WARN [10-03|13:10:26.441] Post-merge network, but no beacon client seen. Please launch one to follow the chain!
```

The above message is emitted when Geth is run without a consensus client on a post-merge proof-of-stake network. 
Since Ethereum moved to proof-of-stake Geth alone is not enough to follow the chain because the consensus logic 
is now implemented by a separate piece of software called a consensus client. This log message is displayed 
when the consensus client is missing. Read more about this on our 

> [consensus clients](/docs/interface/consensus-clients.md) page.

Reviewer 3: ```suggestion
[consensus clients](/docs/interface/consensus-clients) page.
```

> [consensus clients](/docs/interface/consensus-clients.md) page.

Reviewer 2: Link doesn't work
